flickit-assessment-kit/src/test/java/org/flickit/assessment/kit/application/service/answerrange/UpdateAnswerRangeServiceTest.java

```diff
@@ -0,0 +1,109 @@
+package org.flickit.assessment.kit.application.service.answerrange;
+
+import org.flickit.assessment.common.exception.AccessDeniedException;
+import org.flickit.assessment.common.exception.ValidationException;
+import org.flickit.assessment.kit.application.domain.KitVersion;
+import org.flickit.assessment.kit.application.port.in.answerrange.UpdateAnswerRangeUseCase;
+import org.flickit.assessment.kit.application.port.out.answerrange.UpdateAnswerRangePort;
+import org.flickit.assessment.kit.application.port.out.expertgroup.LoadExpertGroupOwnerPort;
+import org.flickit.assessment.kit.application.port.out.kitversion.LoadKitVersionPort;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import static org.flickit.assessment.common.error.ErrorMessageKey.COMMON_CURRENT_USER_NOT_ALLOWED;
+import static org.flickit.assessment.kit.common.ErrorMessageKey.UPDATE_ANSWER_RANGE_TITLE_NOT_NULL;
+import static org.flickit.assessment.kit.test.fixture.application.AssessmentKitMother.simpleKit;
+import static org.flickit.assessment.kit.test.fixture.application.KitVersionMother.createKitVersion;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class UpdateAnswerRangeServiceTest {
+
+    @InjectMocks
+    private UpdateAnswerRangeService service;
+
+    @Mock
+    private LoadExpertGroupOwnerPort loadExpertGroupOwnerPort;
+
+    @Mock
+    private LoadKitVersionPort loadKitVersionPort;
+
+    @Mock
+    private UpdateAnswerRangePort updateAnswerRangePort;
+
+    UUID ownerId = UUID.randomUUID();
+    private final KitVersion kitVersion = createKitVersion(simpleKit());
+
+    @Test
+    void testUpdateAnswerRange_WhenCurrentUserIsNotExpertGroupOwner_ThenThrowAccessDeniedException() {
+        var param = createParam(UpdateAnswerRangeUseCase.Param.ParamBuilder::build);
+
+        when(loadKitVersionPort.load(param.getKitVersionId())).thenReturn(kitVersion);
+        when(loadExpertGroupOwnerPort.loadOwnerId(kitVersion.getKit().getExpertGroupId())).thenReturn(ownerId);
+
+        var throwable = assertThrows(AccessDeniedException.class, () -> service.updateAnswerRange(param));
+        assertEquals(COMMON_CURRENT_USER_NOT_ALLOWED, throwable.getMessage());
+
+        verifyNoInteractions(updateAnswerRangePort);
+    }
+
+    @Test
+    void testUpdateAnswerRange_WhenCurrentUserIsExpertGroupOwnerAndReusableAnswerRangeTitleIsNull_ThenThrowValidationException() {
+        var param = createParam(b -> {
```

Comment:
can be written this way: 
`createParam(b -> b.currentUserId(ownerId).title(null));`